firebase-installations/src/main/java/com/google/firebase/installations/time/SystemClock.java

```diff
@@ -0,0 +1,29 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.installations.time;
+
+/**
+ * Implementation that uses System.currentTimeMillis() to implement {@link Clock#millis()}.
+ *
+ * @hide
+ */
+public class SystemClock implements Clock {
```

Comment:
This class should probably also be a singleton class, but it's not important.
I leave it up to you. I just wanted to point that out.

Reply:
Oh true. I`ll need a little practice to quickly identify singleton classes.Thanks for pointing it out :) Changed it.